File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.go
func getNoZeroIntegers(n int) []int {
	isNoZero := func(x int) bool {
		for x > 0 {
			if x%10 == 0 {
				return false
			}
			x /= 10
		}
        
		return true
	}

	for a := 1; a < n; a++ {
		b := n - a
		if isNoZero(a) && isNoZero(b) {
			return []int{a, b}
		}
	}

	return []int{}
}

File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> list[int]:
        def is_no_zero(x: int) -> bool:
            return '0' not in str(x)

        for a in range(1, n):
            b = n - a
            if is_no_zero(a) and is_no_zero(b):
                return [a, b]
        return []
